domain/port/state/state.go

```diff
@@ -211,86 +190,178 @@ AND endpoint IN ($endpoints[:])
 	return endpoints, nil
 }
 
-// getProtocolMap returns a map of protocol names to their IDs in DQLite.
-func (st *State) getProtocolMap(ctx context.Context, tx *sqlair.TX) (map[string]int, error) {
-	getProtocols, err := st.Prepare("SELECT &protocol.* FROM protocol", protocol{})
+// getOpenedPorts returns the opened ports for the given unit.
+//
+// NOTE: This differs from GetOpenedPorts in that it returns port ranges with
+// their UUIDs, which are not needed by GetOpenedPorts.
+func (st *State) getOpenedPorts(ctx context.Context, tx *sqlair.TX, unitUUID unitUUID) ([]endpointPortRangeUUID, error) {
+	getOpenedPorts, err := st.Prepare(`
+SELECT 
+	port_range.uuid AS &endpointPortRangeUUID.uuid,
+	protocol.protocol AS &endpointPortRangeUUID.protocol,
+	port_range.from_port AS &endpointPortRangeUUID.from_port,
+	port_range.to_port AS &endpointPortRangeUUID.to_port,
+	unit_endpoint.endpoint AS &endpointPortRangeUUID.endpoint
+FROM port_range
+JOIN protocol ON port_range.protocol_id = protocol.id
+INNER JOIN unit_endpoint ON port_range.unit_endpoint_uuid = unit_endpoint.uuid
```

Comment:
Mix of implicit (`JOIN`) and explicit (`INNER JOIN`), the latter of which is not required.